### PR TITLE
Amend `check_year_valid` for NSU's

### PR DESCRIPTION
**Pull request by Jennit07**
I've updated the `check_year_valid` function to include years we do not have NSU for. This affected the running process for 22/23 in the latest ep file testing. This should now pick this up when trying to run this year.

### CODE REVIEW

**Review by Moohan — APPROVED** [review]
Looks good to me, and nice work adding all the extra tests. Approving on the assumption that you ran this code as an ep file last week?

**Comment by Jennit07** [comment]
> Looks good to me, and nice work adding all the extra tests. Approving on the assumption that you ran this code as an ep file last week?

I ran the ep code using `year >= "2223' in place of the check year valid. I think this needs another look as it errors when calling `get_nsu_path` for 22/23

**Comment by Moohan** [comment]
> @check-spelling-bot apply [updates](https://github.com/Public-Health-Scotland/source-linkage-files/actions/runs/4981058958/attempts/1).

File: R/add_nsu_cohort.R [hunk 1]
```diff
@@ -11,103 +11,107 @@
...
-    variables = c(
-      "year",
-      "chi",
-      "recid",
-      "smrtype",
-      "postcode",
-      "gpprac",
-      "dob",
-      "gender"
+  if (check_year_valid("2223", "NSU")) {
```
Comment (Moohan) [inline R/add_nsu_cohort.R:14]:
```suggestion
  if (check_year_valid(year, "NSU")) {
```

**Comment by Jennit07** [comment]
Applied the suggestions and now ready to re-review and merge @Moohan

**Review by Moohan — APPROVED** [review]

**Pull request merged**